From 52942f70918ff2494c0df4f683bc122f9a8e6b82 Mon Sep 17 00:00:00 2001
From: "Lin, Meng-Bo" <linmengbo0689@protonmail.com>
Date: Thu, 9 Dec 2021 21:53:06 +0900
Subject: [PATCH 197/210] arm64: dts: qcom: msm8916-samsung-a5-zt: Add initial
 device tree

Samsung Galaxy A5 (SM-A500YZ) is a smartphone using the MSM8916 SoC released
in 2015.

The A5 variants are very similar, with some differences in accelerometer and
touch key. The common parts are shared in msm8916-samsung-a5u-eur.dts to reduce
duplication.

The difference of accelerometer between A5U/A5ZT is the way the sensor is
mounted on the mainboard - set the mount-matrix in the
device-specific device tree part to handle that difference.

On A5ZT the touch key is supplied by a single fixed
regulator (enabled via GPIO 60) that supplies both MCU and LED.

A5ZT doesn't have an NFC chip.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>
---
 arch/arm64/boot/dts/qcom/Makefile             |  1 +
 .../boot/dts/qcom/msm8916-samsung-a5-zt.dts   | 30 +++++++++++++++++++
 2 files changed, 31 insertions(+)
 create mode 100644 arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts

diff --git a/arch/arm64/boot/dts/qcom/Makefile b/arch/arm64/boot/dts/qcom/Makefile
index 880230666bc..6616e914201 100644
--- a/arch/arm64/boot/dts/qcom/Makefile
+++ b/arch/arm64/boot/dts/qcom/Makefile
@@ -24,6 +24,7 @@ dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-motorola-surnia.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-mtp.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-oppo-a51f.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a3u-eur.dtb
+dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a5-zt.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a5u-eur.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-gprime.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-gt510.dtb
diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts b/arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts
new file mode 100644
index 00000000000..ae4a2805390
--- /dev/null
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include "msm8916-samsung-a5u-eur.dts"
+
+/ {
+	model = "Samsung Galaxy A5 (SM-A500YZ)";
+	compatible = "samsung,a5-zt", "samsung,a5u-eur", "qcom,msm8916";
+	chassis-type = "handset";
+
+	/* NOTE: a5-zt doesn't actually have an NFC chip. */
+	i2c-nfc {
+		status = "disabled";
+	};
+};
+
+&accelerometer {
+	mount-matrix = "0", "1", "0",
+			"1", "0", "0",
+			"0", "0", "1";
+};
+
+&reg_touch_key {
+	gpio = <&msmgpio 60 GPIO_ACTIVE_HIGH>;
+};
+
+&tkey_en_default {
+	pins = "gpio60";
+};
-- 
2.34.1

